R/checkmate.R

```diff
@@ -0,0 +1,176 @@
+  assert_logical(bounds_closed, min.len = 1, max.len = 2, any.missing = FALSE)
+
+  is_valid <- check_numeric(
+    x,
+    finite = TRUE,
+    any.missing = FALSE,
+    len = len,
+    sorted = sorted
+  )
+
+  if (length(bounds_closed) == 1L) {
+    bounds_closed <- rep(bounds_closed, 2L)
+  }
+
+  if (isTRUE(is_valid)) {
+    lwr <- ifelse(bounds_closed[1], all(x >= 0), all(x > 0))
+    upr <- ifelse(bounds_closed[2], all(x <= 1), all(x < 1))
+    in_bounds <- lwr && upr
+
+    if (!in_bounds) {
+      is_valid <- paste(
+        "Probability must be within",
+        ifelse(bounds_closed[1], "[0,", "(0,"),
+        ifelse(bounds_closed[2], "1]", "1)"),
+        "bounds but it is not"
+      )
+    }
+  }
+
+  is_valid
+}
+
+#' @rdname check_probability
+#' @inheritParams check_probability
+#' @export
+assert_probability <- makeAssertionFunction(check_probability)
+
+#' @rdname check_probability
+#' @export
+test_probability <- makeTestFunction(check_probability)
+
+#' @rdname check_probability
+#' @inheritParams check_probability
+#' @export
+expect_probability <- makeExpectationFunction(check_probability)
+
+# assert_prob ----
+
+#' Check if an Argument is a Single Probability Value
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' Check if a given value represents a probability, that is a number within
+#' (0, 1) interval, that can optionally be closed at any side.
+#'
+#' @param x (`number`)\cr a single value to check.
+#' @inheritParams check_probability
+#'
+#' @return `TRUE` if successful, otherwise a string with the error message.
+#'
+#' @seealso [`assertions`] for more details.
+#'
+#' @export
+#' @examples
+#' check_prob(0.5)
+#' check_prob(0, bounds_closed = FALSE)
+#' check_prob(0, bounds_closed = c(FALSE, TRUE))
+check_prob <- function(x, bounds_closed = TRUE) {
+  check_probability(x = x, bounds_closed = bounds_closed, len = 1)
+}
+
+#' @rdname check_prob
+#' @inheritParams check_prob
+#' @export
+assert_prob <- makeAssertionFunction(check_prob)
+
+#' @rdname check_prob
+#' @export
+test_prob <- makeTestFunction(check_prob)
+
+#' @rdname check_prob
+#' @inheritParams check_prob
+#' @export
+expect_prob <- makeExpectationFunction(check_prob)
+
+# assert_prob_range ----
+
+#' Check if an Argument is a Probability Range
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' Check if a given numerical interval represents a probability range, that is
+#' a sub-interval of (0, 1) interval, that can optionally be closed at any side.
+#'
+#' @param x (`number`)\cr an interval to check.
+#' @inheritParams check_probability
+#'
+#' @return `TRUE` if successful, otherwise a string with the error message.
+#'
+#' @seealso [`assertions`] for more details.
+#'
+#' @export
+#' @examples
+#' x <- c(0, 0.2)
+#' check_prob_range(x)
+#' check_prob_range(rev(x))
+#' check_prob_range(x, bounds_closed = FALSE)
+#' check_prob_range(x, bounds_closed = c(FALSE, TRUE))
+check_prob_range <- function(x, bounds_closed = TRUE) {
```

Comment:
then this can be `check_probability_ranges`

Reply:
Can it be `check_probability_range` are we check just one range?

Reply:
ah yes